### PR TITLE
ci(release): automate release steps for the patch

### PR DESCRIPTION
**Pull request by v1v**
## Motivation/summary

> NOTE: This PR is based on https://github.com/elastic/apm-server/pull/12233

* Delegate the FF/Post Release steps to be defined in the project. 
* Use GitHub actions with a well-defined contract (input parameters).
* Those Actions could be called by the Unified Release automatically using [workflow_dispatch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch).

Please take a look at the test section in the description that explains what PRs and changes have been done when running the make file changes locally.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

After resetting my personal git branch to the same commit

```bash
make patch-release RELEASE_VERSION=8.11.5 PROJECT_OWNER=v1v BASE_BRANCH=feature/patch-release
make patch-release RELEASE_VERSION=8.11.5 PROJECT_OWNER=v1v BASE_BRANCH=feature/patch-release
INFO: Create feature branch and update the versions. Target branch 8.11
::group::create-branch update-8.11.6
Switched to branch '8.11'
Your branch is up to date with 'origin/8.11'.
Switched to a new branch 'update-8.11.6'
::endgroup::
>> update-version
>> update-version-makefile
>> update-version-legacy
::group::git-diff
diff --git a/cmd/intake-receiver/version.go b/cmd/intake-receiver/version.go
index ab1ebd068..33425a033 100644
--- a/cmd/intake-receiver/version.go
+++ b/cmd/intake-receiver/version.go
@@ -18,4 +18,4 @@
 package main
 
 // version matches the APM Server's version
-const version = "8.11.4"
+const version = "8.11.5"
diff --git a/internal/version/version.go b/internal/version/version.go
index 3affce343..a1226b696 100644
--- a/internal/version/version.go
+++ b/internal/version/version.go
@@ -18,4 +18,4 @@
 package version
 
 // Version holds the APM Server version.
-const Version = "8.11.4"
+const Version = "8.11.5"
::endgroup::
::group::create-commit
 M cmd/intake-receiver/version.go
 M internal/version/version.go
[update-8.11.6 3d61b4170] docs: update docs versions to 8.11.5
 2 files changed, 2 insertions(+), 2 deletions(-)
::endgroup::
INFO: Push changes to v1v/apm-server and create the relevant Pull Requests
::group::create-pull-request
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 12 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 1.27 KiB | 216.00 KiB/s, done.
Total 8 (delta 6), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (6/6), completed with 6 local objects.
remote: 
remote: Create a pull request for 'update-8.11.6' on GitHub by visiting:
remote:      https://github.com/v1v/apm-server/pull/new/update-8.11.6
remote: 
To https://github.com/v1v/apm-server.git
 * [new branch]          update-8.11.6 -> update-8.11.6

Creating pull request for update-8.11.6 into 8.11 in v1v/apm-server

https://github.com/v1v/apm-server/pull/34
::endgroup::

```

produced https://github.com/v1v/apm-server/pull/34

## Related issues

Requires https://github.com/elastic/apm-server/pull/12233

### CODE REVIEW

File: release.mk [hunk 1]
```diff
@@ -107,18 +109,24 @@ minor-release:
...
-	@echo "VERSION: $${RELEASE_VERSION}"
-	@echo 'TODO: prepare-patch-release'
-	@echo 'TODO: create-prs-patch-release'
+	@echo "INFO: Create feature branch and update the versions. Target branch $(RELEASE_BRANCH)"
+	$(MAKE) create-branch NAME=update-$(NEXT_RELEASE) BASE=$(RELEASE_BRANCH)
+	$(MAKE) update-version VERSION=$(RELEASE_VERSION)
+	$(MAKE) update-version-makefile VERSION=$(PROJECT_MAJOR_VERSION)\.$(PROJECT_MINOR_VERSION)
+	$(MAKE) update-version-legacy VERSION=$(NEXT_RELEASE) PREVIOUS_VERSION=$(CURRENT_RELEASE)
+	$(MAKE) create-commit COMMIT_MESSAGE="docs: update docs versions to $(RELEASE_VERSION)"
+
```
Comment (v1v) [inline release.mk:127]:
```suggestion
```

**Review by amannocci — DISMISSED** [review]
LGTM

File: release.mk [hunk 2]
```diff
@@ -107,18 +109,23 @@ minor-release:
...
 patch-release:
-	@echo "VERSION: $${RELEASE_VERSION}"
-	@echo 'TODO: prepare-patch-release'
-	@echo 'TODO: create-prs-patch-release'
+	@echo "INFO: Create feature branch and update the versions. Target branch $(RELEASE_BRANCH)"
+	$(MAKE) create-branch NAME=update-$(NEXT_RELEASE) BASE=$(RELEASE_BRANCH)
+	$(MAKE) update-version VERSION=$(RELEASE_VERSION)
+	$(MAKE) update-version-makefile VERSION=$(PROJECT_MAJOR_VERSION)\.$(PROJECT_MINOR_VERSION)
+	$(MAKE) update-version-legacy VERSION=$(NEXT_RELEASE) PREVIOUS_VERSION=$(CURRENT_RELEASE)
+	$(MAKE) create-commit COMMIT_MESSAGE="docs: update docs versions to $(RELEASE_VERSION)"
```
Comment (carsonip) [inline release.mk:126]:
```suggestion
	$(MAKE) create-commit COMMIT_MESSAGE="docs: update docs versions to $(NEXT_RELEASE)"
```

File: release.mk [hunk 3]
```diff
@@ -107,18 +109,23 @@ minor-release:
...
+	@echo "INFO: Push changes to $(PROJECT_OWNER)/apm-server and create the relevant Pull Requests"
+	$(MAKE) create-pull-request BRANCH=update-$(NEXT_RELEASE) TARGET_BRANCH=$(RELEASE_BRANCH) TITLE="$(RELEASE_VERSION): update docs" BODY="Merge before the final Release build."
```
Comment (carsonip) [inline release.mk:128]:
```suggestion
	$(MAKE) create-pull-request BRANCH=update-$(NEXT_RELEASE) TARGET_BRANCH=$(RELEASE_BRANCH) TITLE="$(NEXT_RELEASE): update docs" BODY="Merge before the final Release build."
```

**Review by carsonip — COMMENTED** [review]
IIUC these are needed to align PR subject and description with actual changes.

**Review by carsonip — APPROVED** [review]

**Pull request merged**